OneCommunityGlobal/HighestGoodNetworkApp
# Yubo fix obvious non editable field in task view

# Description
<img width="714" alt="Frontend_Task143_PR_description" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/bec062f8-9fb6-4faf-b33a-97aa3d7c4dcd">

Fixes 16 in **PROJECTS/TASKS/WBS COMPONENT** (medium)

## Main changes explained:

1. Refactor TaskEditModal to some small components to add **Edit** permission. Now in Task View, users who only have **view** permission (e.g. Volunteer user) can only see the contents rather than some inputs .
2. Add a edit permission to **TagsSearch** component to select if to render the input.

## How to test:

check into current branch
do `npm install` and `npm run start:local` to run this PR locally

log as **volunteer** user
go to dashboard→ Click any task in **Task** tab -> click **View** button to see the details of each task.
The fields that **volunteer** can't edit should be obvious.

Also log as **admin** or **manager** , they have either **edit** or **suggest** permission, so the fields can be edited and has `update` button. Also try if **Add Task** button can show the Modal properly.

## Screenshots or videos of changes:
<img width="962" alt="Frontend_PR860_comparing" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/5f68f14e-ce91-40fb-9532-80fcbbec64e1">

### Video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/0a7d453a-c5ce-4e4d-961b-c74396f4c8b0




## Note !!! IMPORTANT:
1. Some redundant functions are removed in PR #918 after merged (JUL 8)
2. . If you find such errors like "receive NaN" or "input can't be null", That is due to `calHoursEstimate()` function parse an empty string and it's now fixed (updated: JUN 18).


<img width="673" alt="Screenshot 2023-06-10 at 1 44 01 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/9a6c9db0-75d4-49bd-9bd8-be4db58caae9">
